### PR TITLE
Fix a debug message under MPIR_Allreduce_intra_ring

### PR DESCRIPTION
**Pull request by sagarth**
## Pull Request Description

This PR has two fixes for MPIR_Allreduce_intra_ring:
- Fix an incorrectly composed debug message 
- Adapt the debug statement to a recent datatype change for few parameters from int to MPI_Aint 

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [ ] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: src/mpi/coll/allreduce/allreduce_intra_ring.c [hunk 1]
```diff
@@ -126,7 +126,7 @@ int MPIR_Allreduce_intra_ring(const void *sendbuf, void *recvbuf, MPI_Aint count
 
         MPL_DBG_MSG_FMT(MPIR_DBG_COLL, VERBOSE,
                         (MPL_DBG_FDEST,
-                         "displs[recv_rank:%d]:%d, cnts[recv_rank:%d, displs[send_rank:%d]:%d, cnts[send_rank:%d]:%d]:%d ",
+                         "displs[recv_rank:%d]:%ld, cnts[recv_rank:%d]:%ld, displs[send_rank:%d]:%ld, cnts[send_rank:%d]:%ld ",
                          recv_rank, displs[recv_rank], recv_rank, cnts[recv_rank], send_rank,
                          displs[send_rank], send_rank, cnts[send_rank]));
```
Comment (hzhou) [inline src/mpi/coll/allreduce/allreduce_intra_ring.c:131]: `%ld` is not the correct format for `MPI_Aint`, for example on 32-bit systems, or if `MPI_Aint` is long long.
Can we just remove these debug statements? It is not worth the effort maintaining them.

Comment (sagarth) [reply src/mpi/coll/allreduce/allreduce_intra_ring.c:131]: @hzhou Do you want to remove this debug message in your active PR https://github.com/pmodels/mpich/pull/5940?  As they fall under the same theme.

Comment (hzhou) [reply src/mpi/coll/allreduce/allreduce_intra_ring.c:131]: I prefer your team does it :)

Comment (sagarth) [reply src/mpi/coll/allreduce/allreduce_intra_ring.c:131]: Okay, I can make that change in this PR then. :)

**Review by hzhou — APPROVED** [review]
Great!

**Comment by hzhou** [comment]
Fix the spelling :)

**Review by raffenet — APPROVED** [review]

**Comment by sagarth** [comment]
Rebased and all checks passed.

**Pull request merged**